File: from-java-to-kotlin/src/main/java/pl/braintelligence/E01_String.java
package pl.braintelligence;

public class E01_String {
    public static void main(String[] args) {
        final int a = 5;
        final int b = 7;

        System.out.println(
                getStringNormalConcat(a, b)
        );

        System.out.println(
                getStringFasterPerformanceConcat(a, b)
        );

        System.out.println(
                getMultilineText()
        );

    }

    private static String getStringNormalConcat(int a, int b) {
        return "sum of " + a + " and " + b + " is " + a + b;
    }

    private static String getStringFasterPerformanceConcat(int a, int b) {
        return "sum of "
                .concat(String.valueOf(a))
                .concat(" and ")
                .concat(String.valueOf(b))
                .concat(" is ")
                .concat(String.valueOf(a + b));
    }

    private static String getMultilineText() {
        return "First Line\n" +
                "Second Line\n" +
                "Third Line";
    }
}

File: from-java-to-kotlin/src/main/java/pl/braintelligence/E01_String.kt
package pl.braintelligence

fun main() {
    val a = 4
    val b = 7

    // Java way
    println(getStringInterpolation(a, b))

    // Kotlin way - String interpolation
    getStringInterpolation(1, 1).run { print(this) }

    // Kotlin way - multiline String
    getMultiline().run { print(this) }
}

// without class function exists in package scope
fun getStringInterpolation(a: Int, b: Int) = "sum of $a and $b is ${a + b}"

fun getMultiline() = """
        |First Line
        |Second Line
        |Third Line
    """.trimMargin()

